src/lib/assetBridger/erc20Bridger.ts

```diff
@@ -884,6 +885,31 @@ export class AdminErc20Bridger extends Erc20Bridger {
     await l1Token.deployed()
     await l2Token.deployed()
 
+    if (!this.nativeTokenIsEth) {
+      const nativeTokenContract = ERC20__factory.connect(
+        this.nativeToken!,
+        l1Provider
+      )
+      const allowance = await nativeTokenContract.allowance(
+        l1SenderAddress,
+        l1Token.address
+      )
+
+      const gEstimator = new L1ToL2MessageGasEstimator(l2Provider)
+
+      // increase by 500%
+      const maxFeePerGas = (await gEstimator.estimateMaxFeePerGas()).mul(6)
+
+      // hardcode gas limit to 60k
+      const estimatedGas = BigNumber.from(60_000).mul(maxFeePerGas)
+
+      if (allowance.lt(estimatedGas)) {
```

Comment:
you don't neeed to use `L1ToL2MessageGasEstimator` + let's use `percentIncrease` (we can copy it over from `L1ToL2MessageGasEstimator` https://github.com/OffchainLabs/arbitrum-sdk/blob/main/src/lib/message/L1ToL2MessageGasEstimator.ts#L76)

```suggestion
      const maxFeePerGasOnL2 = (await l2Provider.getFeeData()).maxFeePerGas
      const maxFeePerGasOnL2WithBuffer = percentIncrease(maxFeePerGasOnL2, 500)

      // hardcode gas limit to 60k
      const estimatedGasFee = BigNumber.from(60_000).mul(maxFeePerGasOnL2WithBuffer)

      if (allowance.lt(estimatedGasFee)) {
```